### PR TITLE
docs: add property-based scheduling concept docs

### PR DESCRIPTION
**Pull request by michaelawyu**
### Description of your changes

This PR adds a concept documentation piece for property-based scheduling.

I have:

- [x] Run `make reviewable` to ensure this PR is ready for review.

### How has this code been tested

N/A

### Special notes for your reviewer

N/A

### CODE REVIEW

File: docs/concepts/PropertyProviderAndClusterProperties/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+	// This call should not block.
+	//
+	// Note that Fleet member agent will cancel the context when it exits.
+	Start(ctx context.Context, config *rest.Config) error
+}
+```
+
+For the details, see the [Fleet source code](../../../pkg/propertyprovider/interface.go).
+
+A property provider should be shipped as a part of the Fleet member agent and runs alongside it.
```
Comment (kaysieyu) [inline docs/concepts/PropertyProviderAndClusterProperties/README.md:38]:
```suggestion
A property provider should be shipped as a part of the Fleet member agent and run alongside it.
```

File: docs/concepts/PropertyProviderAndClusterProperties/README.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+The Fleet member agent can run with or without a property provider. If a provider is not set up, or
+the given provider fails to start properly, the agent will collect limited properties about
+the cluster on its own, specifically the total and allocatable CPU and memory capacities of
+the host member cluster. 
+
+## Cluster properties
+
+A cluster property is an attribute of a member cluster. There are two types of properties:
+
+* A resource property describes the usage information of a resource in a member cluster; the
```
Comment (kaysieyu) [inline docs/concepts/PropertyProviderAndClusterProperties/README.md:60]:
```suggestion
* Resource Property: describes the usage information of a resource in a member cluster; the
```

File: docs/concepts/PropertyProviderAndClusterProperties/README.md [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    installed in the cluster;
+    * the allocatable capacity of the resource, which is the maximum amount of the resource 
+    that can be used for running user workloads, as some amount of the resource might be
+    reserved by the OS, kubelet, etc.;
+    * the available capacity of the resource, which is the amount of the resource that
+    is currently free for running user workloads.
+
+    Note that you may report a virtual resource via the property provider, if applicable.
+
+* A non-resource property is essentially a metric about a member cluster, in the form of a key/value
```
Comment (kaysieyu) [inline docs/concepts/PropertyProviderAndClusterProperties/README.md:75]:
```suggestion
* Non-resource property: a metric about a member cluster in the form of a key/value
```

File: docs/concepts/PropertyProviderAndClusterProperties/README.md (hunk 1 above)
Comment (michaelawyu) [reply docs/concepts/PropertyProviderAndClusterProperties/README.md:38]: Done.

File: docs/concepts/PropertyProviderAndClusterProperties/README.md (hunk 2 above)
Comment (michaelawyu) [reply docs/concepts/PropertyProviderAndClusterProperties/README.md]: Done.

File: docs/concepts/PropertyProviderAndClusterProperties/README.md (hunk 3 above)
Comment (michaelawyu) [reply docs/concepts/PropertyProviderAndClusterProperties/README.md]: Done.

**Review by ryanzhang-oss — APPROVED** [review]

**Pull request merged**